juju/juju
# Generalize provider tracker

This generalizes the provider trackers to handle both providers and caas brokers in the same 1 worker. Both providers and brokers require watching the same information; the cloud and credential changes, along with updating the cloud spec.

Integration in the dependency engine and removal of the workers can then be aligned to 1 worker servicing multiple users. Removal of the caas broker can't be fully done, as the jujud model operator requires access to the broker, which will do so over the API connection.

A follow-up to this PR will create a runner inside the worker that allows the caching of multiple environs per model uuid.

----

## Checklist




## QA steps

Ensure that we can bootstrap both IAAS and CAAS.

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```

```sh
$ juju bootstrap microk8s test
$ juju add-model default
$ juju deploy redis-k8s
```


## Links

**Jira card:** JUJU-5643
